### ISSUE TITLE
caffe2 softmaxwithloss problem

### ISSUE THREAD

**Issue by nebuladream** [pytorch/pytorch#16494]
my output label is  batch_size x  10 （for 10 labels）
 blob_out = model.FC(  blob_out, 'pred', dim_in, 10)
softmax, loss = model.SoftmaxWithLoss(  [blob_out, labels], ['softmax', 'loss'], scale=scale)   
 but SoftmaxWithLoss report error like this:
 what():  [enforce fail at softmax_ops.cu:307] T.size_from_dim(canonical_axis) == 1. 10 vs 1 Error from operator: 
input: "gpu_0/pred" input: "gpu_0/labels" output: "gpu_0/softmax" output: "gpu_0/loss" name: "" type: "SoftmaxWithLoss" arg { name: "scale" f: 1 } device_option { device_type: 1 cuda_gpu_id: 0 }
0 0 0 1 0 0 0 1 0 0 *** Aborted at 1548750491 (unix time) try "date -d @1548750491" if you are using GNU date ***
PC: @     0x7f89558a1428 gsignal

**Comment by nebuladream** [comment]
I define the input op as:
label_shape[0] = batch_size_;
label_shape[1] = 10;
prefetched_label_.Resize(label_shape);
